cpp/src/gandiva/precompiled/timestamp_arithmetic.cc

```diff
@@ -96,58 +96,70 @@ extern "C" {
 
 TIMESTAMP_DIFF(timestamp)
 
-#define ADD_INT32_TO_TIMESTAMP_FIXED_UNITS(TYPE, NAME, TO_MILLIS) \
-  FORCE_INLINE                                                    \
-  TYPE NAME##_int32_##TYPE(int32 count, TYPE millis) {            \
-    return millis + TO_MILLIS * static_cast<TYPE>(count);         \
-  }
-
 // Documentation of mktime suggests that it handles
 // TmMon() being negative, and also TmMon() being >= 12 by
 // adjusting TmYear() accordingly
 //
 // Using gmtime_r() and timegm() instead of localtime_r() and mktime()
 // since the input millis are since epoch
-#define ADD_INT32_TO_TIMESTAMP_MONTH_UNITS(TYPE, NAME, N_MONTHS)                \
```

Comment:
This rename doesn't make sense to me; it's used to generate functions like `timestampaddMonth_int32_timestamp` so why should the name match `/(timestamp).*(int32)/i` instead of `/(int32).*(timestamp)/i`?

Reply:
this is related to [73ceae41174f70ed40f](https://github.com/apache/arrow/commit/73ceae41174f70ed40f5dc5b5365e112727f1567). some of the changes didn't make sense so I reverted them.

Reply:
That commit doesn't touch this macro from what I can see

Reply:
body of this macro has changed

Reply:
I don't observe that the body of this macro has changed in this PR, do you mean in #4539 ?

- Prior to #4539 `ADD_INT32_TO_TIMESTAMP_MONTH_UNITS  generates  timestampaddMonth_timestamp_int32`
- currently `ADD_INT32_TO_TIMESTAMP_MONTH_UNITS  generates  timestampaddMonth_int32_timestamp`, which makes the most sense to me as the macro name matches the name of the generated function
- after #4821 `ADD_TIMESTAMP_TO_INT32_MONTH_UNITS  generates  timestampaddMonth_int32_timestamp`